src/components/vulnerabilities/Form.js

```diff
@@ -36,6 +40,8 @@ const VulnerabilityForm = ({
             .then(([projects, categories]) => {
                 const defaultProjectId = projects.length ? projects[0].id : 0;
                 const projectId = isEditForm ? vulnerability.project_id : defaultProjectId;
+                setMetrics(isOwaspProject(projects, projectId))
```

Comment:
This line has one consequence which I cannot decide if its bug or not. When we are creating a new vulnerability, it mimics the "default" project. So, if the first project is owasp, we display the owasp fields.
Once the vulnerability is created and assigned to the project, we use projects values.

Reply:
I understand why you are confused. The form code for this is overly complicated. I need to tidy up but for now the logic that you added is OK. Thanks!